xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/user/impl/xwiki/MyFormAuthenticator.java

```diff
@@ -292,4 +315,45 @@ public boolean processLogout(SecurityRequestWrapper securityRequestWrapper,
         }
         return result;
     }
+
+    private ObservationManager getObservationManager() {
```

Comment:
There is a lot of duplication, seems to me we would need to put all that in a component used by both MyFormAuthenticator an MyBasicAuthenticator.